packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -551,7 +554,12 @@ export abstract class RestApiBase extends Resource implements IRestApi {
         ...props.deployOptions,
       });
 
-      new CfnOutput(this, 'Endpoint', { exportName: props.endpointExportName, value: this.urlForPath() });
+      // If the context is true, don't create an unnamed output.
+      // If false - or the endpointExportName is set, do create a CfnOutput.
+      const avoidEmptyExport: boolean | undefined = Node.of(this).tryGetContext(APIGATEWAY_REST_API_NO_CFN_OUTPUT);
```

Comment:
I believe you should use the `FeatureFlags` API - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/core/lib/feature-flags.ts#L11 - instead.

Reply:
Ooh, nice spot!